runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -257,15 +266,13 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		{
 			type -= MENU_ACTION_DEPRIORITIZE_OFFSET;
 		}
-
-		if (config.highlightMenuNames() &&
-			NPC_MENU_ACTIONS.contains(MenuAction.of(type)) &&
-			highlightedNpcs.stream().anyMatch(npc -> npc.getIndex() == event.getIdentifier()))
+		if (NPC_MENU_ACTIONS.contains(MenuAction.of(type)))
 		{
 			MenuEntry[] menuEntries = client.getMenuEntries();
 			final MenuEntry menuEntry = menuEntries[menuEntries.length - 1];
-			final String target = ColorUtil.prependColorTag(Text.removeTags(event.getTarget()), config.getHighlightColor());
-			menuEntry.setTarget(target);
+			final NPC menuNpc = client.getCachedNPCs()[event.getIdentifier()];
```

Comment:
This is only ever used as an argument to `buildRecoloredNpcMenuEntryTarget`, declare it as a field there and remove the second parameter from the method.